worker/worker.go

```diff
@@ -0,0 +1,39 @@
+package main
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"os/signal"
+	"syscall"
+)
+
+var (
+	// workerID is the id assigned by the scheduler
+	// after registering on scheduler.
+	workerID string
+)
+
+func init() {
+	loadConfig()
+}
+
+// Entry point of the worker application.
+func main() {
+	go registerWorker()
+
+	sig := make(chan os.Signal)
+	signal.Notify(sig, syscall.SIGINT, syscall.SIGTERM)
+
+	for {
+		select {
+		case s := <-sig:
```

Comment:
Nitpick: I'm sure it's for future proofing, but I don't think the for/select is necessary here, since you can just block on the channel read until you have something else you need to hand'e in the select.

Reply:
You are right, the `sig` channel is already blocking. I am removing `for` loops, it is a bit cleaner